h2o-automl/src/main/java/ai/h2o/automl/targetencoding/TargetEncoderModel.java

```diff
@@ -109,23 +111,34 @@ public int nfeatures() {
    * Transform with noise 
    */
   public Frame transform(Frame data, byte strategy, double noiseLevel, long seed){
-    return _targetEncoder.applyTargetEncoding(data, _parms._response_column, this._output._target_encoding_map, strategy,
-            _parms._teFoldColumnName, _parms._withBlending, noiseLevel, true, seed);
+    final TargetEncoder.DataLeakageHandlingStrategy leakageHandlingStrategy = TargetEncoder.DataLeakageHandlingStrategy.fromVal(strategy);
+    return _targetEncoder.applyTargetEncoding(data, _parms._response_column, this._output._target_encoding_map, leakageHandlingStrategy,
+            _parms._teFoldColumnName, _parms._withBlending, noiseLevel,false, seed);
   }
 
   /**
    * Transform with default noise of 0.01 
    */
   public Frame transform(Frame data, byte strategy, long seed){
-    return _targetEncoder.applyTargetEncoding(data, _parms._response_column, this._output._target_encoding_map, strategy,
-            _parms._teFoldColumnName, _parms._withBlending, true, seed);
+
+    final TargetEncoder.DataLeakageHandlingStrategy leakageHandlingStrategy = TargetEncoder.DataLeakageHandlingStrategy.fromVal(strategy);
+    return _targetEncoder.applyTargetEncoding(data, _parms._response_column, this._output._target_encoding_map, leakageHandlingStrategy,
+            _parms._teFoldColumnName, _parms._withBlending,false, seed);
   }
   
   @Override
   protected double[] score0(double data[], double preds[]){
     throw new UnsupportedOperationException("TargetEncoderModel doesn't support scoring. Use `transform()` instead.");
   }
 
+  @Override
+  public Frame score(Frame fr, String destination_key, Job j, boolean computeMetrics, CFuncRef customMetricFunc) throws IllegalArgumentException {
+    final TargetEncoder.DataLeakageHandlingStrategy leakageHandlingStrategy = TargetEncoder.DataLeakageHandlingStrategy.valueOf(_parms._leakageHandlingStrategy);
```

Comment:
where is the frame adapted?

Reply:
This is indeed a bug. I wrongly thought TargetEncoder is going to do the adaptation itself. Or more precisely, it will do the validation on it's own. 

But it is indeed advantageous to do to he adaptation.

Reply:
Nowehere. A bug. Adaptation should indeed be done. Adding adaptation revealed some other bugs.

Reply:
Adaptation may remove some columns from the remaining dataset, which might be useful for making predictions, but not for transformations like target encoding, where necessary predecessor is the dataset having the required features in place.

I'm also suspicious some features of the adaptation process, like categorical levels matching might even hurt the process of Target Encoding. This is all up to the `TargetEncoder` class, which should do the validation.